operator/pkg/ingress/ingress.go

```diff
@@ -0,0 +1,447 @@
+	}
+
+	// check if the endpoints resource already exists
+	_, exists, err := ic.endpointManager.getByKey(key)
+	if err != nil {
+		log.WithError(err).Warn("endpoints lookup returned an error")
+		return err
+	}
+	if exists {
+		// Endpoints already exists in the cache. For now assume that it was created by the ingress
+		// controller.
+		log.WithField("endpoints", key).Info("Endpoints already exists. Continuing...")
+		return nil
+	}
+
+	_, err = k8s.Client().CoreV1().Endpoints(ingress.Namespace).Create(context.Background(), endpoints, metav1.CreateOptions{})
+	if err != nil {
+		log.WithError(err).WithField("ingress", ingress.Name).Error("Failed to create endpoints for ingress")
+		return err
+	}
+
+	log.WithField("endpoints", key).Info("Created Endpoints for Ingress")
+	return nil
+}
+
+func (ic *IngressController) createEnvoyConfig(ingress *slim_networkingv1.Ingress) error {
+	desired, err := getEnvoyConfigForIngress(k8s.Client(), ingress)
+	if err != nil {
+		return err
+	}
+
+	// check if the CiliumEnvoyConfig resource already exists
+	key, err := cache.MetaNamespaceKeyFunc(desired)
+	if err != nil {
+		log.WithError(err).Warn("MetaNamespaceKeyFunc returned an error")
+		return err
+	}
+	existingEnvoyConfig, exists, err := ic.envoyConfigManager.getByKey(key)
+	if err != nil {
+		log.WithError(err).Warn("CiliumEnvoyConfig lookup failed")
+		return err
+	}
+	if exists {
+		if desired.DeepEqual(existingEnvoyConfig) {
+			log.WithField("cilium-envoy-config", key).Info("No change for existing CiliumEnvoyConfig")
+			return nil
+		}
+		// Update existing CEC
+		newEnvoyConfig := existingEnvoyConfig.DeepCopy()
+		newEnvoyConfig.Spec = desired.Spec
+		_, err = k8s.CiliumClient().CiliumV2alpha1().CiliumEnvoyConfigs(ingress.Namespace).Update(context.Background(), newEnvoyConfig, metav1.UpdateOptions{})
+		if err != nil {
+			log.WithError(err).WithField("ingress", ingress.Name).Error("Failed to update CiliumEnvoyConfig for ingress")
+		}
+		log.WithField("ingress", ingress.Name).Error("Updated CiliumEnvoyConfig for ingress")
+		return nil
+	}
+	_, err = k8s.CiliumClient().CiliumV2alpha1().CiliumEnvoyConfigs(ingress.Namespace).Create(context.Background(), desired, metav1.CreateOptions{})
+	if err != nil {
+		log.WithError(err).WithField("ingress", ingress.Name).Error("Failed to create CiliumEnvoyConfig for ingress")
+		return err
+	}
+	log.WithField("ingress", ingress.Name).Info("Create CiliumEnvoyConfig for ingress")
+	return nil
+}
+
+func (ic *IngressController) handleAddIngress(obj interface{}) {
+	if ingress := k8s.ObjToV1Ingress(obj); ingress != nil {
+		ic.queue.Add(ingressAddedEvent{ingress: ingress})
+	}
+}
+
+func (ic *IngressController) handleUpdateIngress(oldObj, newObj interface{}) {
+	oldIngress := k8s.ObjToV1Ingress(oldObj)
+	if oldIngress == nil {
+		return
+	}
+	newIngress := k8s.ObjToV1Ingress(newObj)
+	if newIngress == nil {
+		return
+	}
+	ic.queue.Add(ingressUpdatedEvent{oldIngress: oldIngress, newIngress: newIngress})
```

Comment:
even before adding something into the queue we could do a DeepEqual and if the objects are equal then we can ignore this event.